### PR TITLE
refactor: clean up revisions test

### PR DESCRIPTION
**Pull request by Bento007**
### Reviewers
**Functional:* @MDunitz 

**Readability:** 

---

## Changes
- moving coming test code into BaseRevisionTest

### CODE REVIEW

File: tests/unit/backend/chalice/api_server/test_revisions.py [hunk 1]
```diff
@@ -232,57 +231,82 @@ def test__delete_published_dataset_during_revision(self):
...
-        resp.raise_for_status()
-        self.assertEqual(json.loads(resp.body)["datasets"], [])
+class TestPublishRevision(BaseRevisionTest):
+    def setUp(self):
+        super().setUp()
+        self.test_url = f"/dp/v1/collections/{self.pub_collection.id}/publish"
 
-        # Published assets are not deleted
-        self.assertPublishedCollectionOK(expected_body, pub_s3_objects)
+    def publish_collection(self, collection_id: str):
```
Comment (MDunitz) [inline tests/unit/backend/chalice/api_server/test_revisions.py:253]: Would it be worth putting this function in `BaseRevisionTest` or even `BaseAuthAPITest`?

File: tests/unit/backend/chalice/api_server/test_revisions.py [hunk 2]
```diff
@@ -232,57 +231,82 @@ def test__delete_published_dataset_during_revision(self):
...
+        path = f"/dp/v1/collections/{collection_id}"
+        headers = {"host": "localhost", "Content-Type": "application/json"}
+        response = self.app.get(path, headers)
+        response.raise_for_status()
+        actual = json.loads(response.body)
+        self.assertEqual("PUBLIC", actual["visibility"])
+        self.assertEqual(collection_id, actual["id"])
+        return actual
+
+    def verify_datasets(self, actual_body, expected_dataset_ids):
```
Comment (MDunitz) [inline tests/unit/backend/chalice/api_server/test_revisions.py:284]: Would it be worth putting this function in `BaseRevisionTest`

**Review by MDunitz — APPROVED** [review]
lgtm -- made some small suggestions but feel free to ignore

File: tests/unit/backend/chalice/api_server/test_revisions.py (hunk 1 above)
Comment (Bento007) [reply tests/unit/backend/chalice/api_server/test_revisions.py:253]: We can move it once another test needs it.

File: tests/unit/backend/chalice/api_server/test_revisions.py (hunk 2 above)
Comment (Bento007) [reply tests/unit/backend/chalice/api_server/test_revisions.py:284]: We can move it once another test needs it.

**Pull request merged**